core/spec/models/spree/concerns/user_address_book_spec.rb

```diff
@@ -30,37 +30,85 @@ module Spree
           it "sets the UserAddress default flag to true" do
             subject
             expect(user_address.default).to eq true
+            expect(user_address.default_billing).to be_falsey
+          end
+
+          context "billing address" do
+            subject { user.save_in_address_book(address.attributes, true, :billing) }
+
+            it "sets the UserAddress default_billing flag to true" do
+              subject
+              expect(user_address.default).to be_falsey
+              expect(user_address.default_billing).to eq true
+            end
           end
 
           it "adds the address to the user's the associated addresses" do
             expect { subject }.to change { user.reload.addresses.count }.by(1)
           end
         end
 
-        context "user already has a default address" do
+        context "user already has default addresses" do
           let(:address) { create(:address) }
           let(:original_default_address) { create(:ship_address) }
+          let(:original_default_bill_address) { create(:bill_address) }
           let(:original_user_address) { user.user_addresses.find_first_by_address_values(original_default_address.attributes) }
+          let(:original_user_bill_address) { user.user_addresses.find_first_by_address_values(original_default_bill_address.attributes) }
 
           before do
             user.user_addresses.create(address: original_default_address, default: true)
+            user.user_addresses.create(address: original_default_bill_address, default_billing: true)
           end
 
-          it "makes all the other associated addresses not be the default" do
-            expect { subject }.to change { original_user_address.reload.default }.from(true).to(false)
+          context "saving a shipping address" do
+            context "makes all the other associated shipping addresses not be the default and ignores the billing ones" do
+              it do
+                expect { subject }.to change { original_user_address.reload.default }.from(true).to(false)
```

Comment:
Even here, you can use the one-liner syntax:  

```ruby
it { is_expected.to change { original_user_address.reload.default  }.from(true).to(false) }
```

Reply:
Done